File: springboot/src/main/java/com/example/springboot/entity/Doctor.java
package com.example.springboot.entity; // 包名调整

import com.example.springboot.entity.enums.DoctorStatus; // 导入路径调整
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

/**
 * 医生表
 */
@Entity
@Table(name = "doctors")
@Data
public class Doctor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer doctorId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "clinic_id", nullable = false)
    private Clinic clinic; // 所属校医院/诊所

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id", nullable = false)
    private Department department; // 所属科室

    @Column(nullable = false, unique = true, length = 100)
    private String identifier; // 医生的工号

    @Column(name = "password_hash", nullable = false)
    private String passwordHash; // 哈希加盐后的密码

    @Column(name = "full_name", nullable = false, length = 100)
    private String fullName; // 真实姓名

    @Column(name = "phone_number", unique = true, length = 20)
    private String phoneNumber; // 存储E.164标准格式

    private String title; // 职称 (如：“主治医师”)
    private String specialty; // 擅长领域描述
    private String bio; // 个人简介

    @Column(name = "photo_url")
    private String photoUrl; // 头像照片URL

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private DoctorStatus status; // 账户状态

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt; // 账户创建时间

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt; // 信息最后更新时间
}

File: vue/src/api/doctor.js
import request from '@/utils/request';

/**
 * 获取当前登录医生的详细个人资料
 * (假设后端提供了 /api/doctor/profile 接口，
 * 它会根据token自动返回当前医生的信息)
 */
export function getDoctorProfile() {
    return request({
        url: '/api/doctor/profile', // 假设后端获取个人资料的接口
        method: 'get'
    });
}

/**
 * 更新当前登录医生的个人资料
 * @param {Object} data - 包含要更新字段的对象
 * (e.g., { phoneNumber, bio, photoUrl })
 */
export function updateDoctorProfile(data) {
    return request({
        url: '/api/doctor/profile', // 假设后端更新个人资料的接口
        method: 'put',
        data: data
    });
}